dali/operators/generic/reshape.cc

```diff
@@ -270,7 +287,22 @@ void Reshape<Backend>::CalculateOutputShape(const Workspace &ws) {
       ShapeFromInput(ws.template InputRef<CPUBackend>(1), false);
       break;
     case ShapeSource::None:
-      output_shape_ = input_shape_;
+      if (src_dims_.empty()) {
+        output_shape_ = input_shape_;
+        break;
+      }
+
+      output_shape_.resize(N, src_dims_.size());
+      for (int i = 0; i < N; i++) {
+        for (size_t d = 0; d < src_dims_.size(); d++) {
+          const int src_d = src_dims_[d];
+          output_shape_.tensor_shape_span(i)[d] =
+            src_d == -1 ? 1 : input_shape_.tensor_shape_span(i)[src_d];
+        }
+        DALI_ENFORCE(
+          volume(output_shape_.tensor_shape_span(i)) == volume(input_shape_.tensor_shape_span(i)),
+          make_string(OpName(), ": ``src_dims`` should remove only dimensions equal to 1"));
```

Comment:
This message can be somewhat confusing when using Squeeze (since the user doesn't know that ``src_dims`` exists. Consider rephrasing to "The volume of the new shape should match the one of the original shape. Requested a shape with <new size> elements but the original shape has <old_size> element."